File: .difffusion/modelling.py
import json
import requests

if __name__ == '__main__':
    modelling_file_name = '.difffusion/modelling.json'
    with open(modelling_file_name, "r") as f:
        data = json.load(f)
    # print(data)
    #
    for obj in data:
        resp = requests.post(
            'http://localhost:2024/api/v1/ops',
            headers={ 'alias' : obj['alias'], 'op' : 'BUILD' },
            data=json.dumps(obj),
            )
        print(resp.json())
    simulation_file_name = '.difffusion/simulation.json'
    with open(simulation_file_name, "r") as f:
        data = json.load(f)
    for key in data:
        resp = requests.post(
            'http://localhost:2024/api/v1/ops',
            headers={ 'alias' : key, 'op' : 'BUILD' },
            data=json.dumps(data[key]),
            )
        print(resp.json())
    #
    resp = requests.get(
        'http://localhost:2024/api/v1/ops',
        headers={ 'alias' : 'cube/all_legs', 'op' : 'BUILD' },
        )
    print(resp.json().keys())
    out_file_name = '.difffusion/cube.json'
    with open(out_file_name, 'w', encoding='utf-8') as f:
        json.dump(resp.json(), f, ensure_ascii=False, indent=4)
    #

    

File: .difffusion/modelling.js
const fs = require('fs');
const axios = require('axios');

const modelling_file_name = '.difffusion/json/modelling.json';
fs.readFile(modelling_file_name, 'utf-8', (err, data) => {
    if (err) {
        console.error(err);
        return;
    }
    const modellingData = JSON.parse(data);
    modellingData.forEach(obj => {
        axios.post(
            'http://localhost:2024/api/v1/ops',
            obj,
            {
                headers: { 'alias': obj.alias, 'op': 'BUILD' }
            }
        )
        .then(resp => {
            console.log(resp.data);
        })
        .catch(error => {
            console.error(error);
        });
    });
});

const simulation_file_name = '.difffusion/json/simulation.json';
fs.readFile(simulation_file_name, 'utf-8', (err, data) => {
    if (err) {
        console.error(err);
        return;
    }
    const simulationData = JSON.parse(data);
    Object.keys(simulationData).forEach(key => {
        axios.post(
            'http://localhost:2024/api/v1/ops',
            simulationData[key],
            {
                headers: { 'alias': key, 'op': 'BUILD' }
            }
        )
        .then(resp => {
            console.log(resp.data);
        })
        .catch(error => {
            console.error(error);
        });
    });
});

axios.get(
    'http://localhost:2024/api/v1/ops',
    {
        headers: { 'alias': 'cube/all_legs', 'op': 'BUILD' }
    }
)
.then(resp => {
    console.log(Object.keys(resp.data));
    const out_file_name = '.difffusion/cube.json';
    fs.writeFile(out_file_name, JSON.stringify(resp.data, null, 4), 'utf-8', (err) => {
        if (err) {
            console.error(err);
            return;
        }
        console.log('File saved successfully');
    });
})
.catch(error => {
    console.error(error);
});

